### PR TITLE
Test for 4 body helicity angles

### LINKED ISSUE #18: [Test] add test for helicity angles
there is a nice method of computing the helicity angles in the packages already.
Is it properly tested?

We can implement a test that validates it using an example from Lc->pKpi,

three different chains:
- Kpi `((2,3),1)`,
- pK `((1,2),3)`, and
- pip `((3,1),2)`.

The helicity angles are given below,
```
    "chain_variables": {
        "Kpi": {
            "mkpisq": 1.3743747462964881,
            "theta_Kst": 1.0468159811504423,
            "phi_Kst": 1.3921357860994747,
            "theta_K": 1.692234518478623,
            "phi_K": 0.5466265769529981
        },
        "pK": {
            "mkpsq": 2.756020646168232,
            "theta_L": 2.6621627135924624,
            "phi_L": 3.010596711405015,
            "Ltheta_p": 1.861461784272743,
            "Lphi_p": 1.3499280354237881
        },
        "pip": {
            "mppisq": 2.2410542593352796,
            "theta_D": 1.05685191949046,
            "phi_D": -1.1654157065810633,
            "theta_pi": 1.1669778470524175,
            "phi_pi": 2.5984404692152796
        }
```

sadly, I did not save the four-vectors, but one reconstruct them by building aligned kinematics, and rotating it back to, e.g.

(in a separate notebook, just to find the four-vectors for input of the test)
1. build align kinematics, from [here](https://github.com/KaiHabermann/decayangle/issues/1#issuecomment-2016925558)
using the Kpi reference orientation (`Pp = [0,0,-lambda...,Ep`])
2. Rotate aligned four-vectors with 
```
Rz(phi_Kphi)*Ry(theta_Kst)*Rz(phi_K)
```

The test would be to get all four angles right for every topology.

### PR DESCRIPTION
**Pull request by KaiHabermann**
Closes #18

### CODE REVIEW

File: tests/test_lorentz.py [hunk 1]
```diff
@@ -283,7 +283,93 @@ def cos_theta_hat_2_canonical_3(M, m1, m2, m3, sigma1, sigma2, sigma3):
...
+    m31sq = m0sq + m1sq + m2sq + m3sq - m12sq - m23sq
+
+
+    # Momenta magnitudes
+    p1a = np.sqrt(Kallen(m23sq, m1sq, m0sq)) / (2*m0)
+    p2a = np.sqrt(Kallen(m31sq, m2sq, m0sq)) / (2*m0)
+    p3a = np.sqrt(Kallen(m12sq, m3sq, m0sq)) / (2*m0)
+
+    # Directions and components
+    cos_zeta_12_for0 = 1.0 # Placeholder for actual expression
```
Comment (mmikhasenko) [inline tests/test_lorentz.py:337]: that is the only problem! Let's get correct expression there

Comment (KaiHabermann) [reply tests/test_lorentz.py]: God damnit
I knew it was something like this, but I couldnt spot it :D

Comment (KaiHabermann) [reply tests/test_lorentz.py]: Thanks

**Comment by KaiHabermann** [comment]
So now your plan is to rotate what exactly? The dict have 4 angles per chain. 
But I dont know what they mean really

**Comment by mmikhasenko** [comment]
p(1)K(2)pi(3), we take `[[2,3],1]` kinematics as reference

```
            "mkpisq": 1.3743747462964881,
            "theta_Kst": 1.0468159811504423,
            "phi_Kst": 1.3921357860994747,
            "theta_K": 1.692234518478623,
            "phi_K": 0.5466265769529981
```

rotations to be applied to the aligned kinematics are

```
Rz(phi_Kst) * Ry(theta_Kst) * Rz(phi_K) p_aligned
``` 

this way you compute `p1`, `p2`, `p3`, that we are going to pass to your `angles` function

**Comment by KaiHabermann** [comment]
Should work now.
@mmikhasenko could you have a quick look?

File: tests/test_lorentz.py [hunk 2]
```diff
@@ -283,7 +294,167 @@ def cos_theta_hat_2_canonical_3(M, m1, m2, m3, sigma1, sigma2, sigma3):
...
+        np.cos(tree3.helicity_angles(momenta_23_rotated)[(1, 2)].psi_rf),
+        np.cos(chain_vars["pK"]["Lphi_p"]),
+        1e-4,
+    )
+    assert np.allclose(
+        np.cos(tree3.helicity_angles(momenta_23_rotated)[(1, 2)].theta_rf),
+        np.cos(chain_vars["pK"]["Ltheta_p"]),
+        1e-4,
+    )
+
```
Comment (KaiHabermann) [inline tests/test_lorentz.py:455]: This is the actual new test. 
Sorry i smuggeled some cleanups in theis PR aswell. 
I used the direct definition of Topologies here, so we would not be bothered by translating dpd definitions into what decayangle does normally

**Pull request merged**